perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -533,18 +548,18 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 		Set<Member> members = new HashSet<>(jdbc.query("select distinct " + memberMappingSelectQuery +
 				" from members " +
 				" left join users u on members.user_id=u.id " +
-				" left join member_attr_values mav1 on members.id=mav1.member_id and mav1.attr_id in (select id from attr_names where attr_name='" + MembersManagerImpl.A_D_MEMBER_MAIl + "') " +
-				" left join user_attr_values uav1 on u.id=uav1.user_id and uav1.attr_id in (select id from attr_names where attr_name='" + MembersManagerImpl.A_D_USER_PREFERRED_MAIL + "') " +
-				" left join user_attr_values uav2 on u.id=uav2.user_id and uav2.attr_id in (select id from attr_names where friendly_name like 'login-namespace:%') " +
+				" left join member_attr_values mav on members.id=mav.member_id and mav.attr_id in (select id from attr_names where attr_name in (" + memberAttributes.toString() + "))" +
```

Comment:
There is possible SQL injection, since you put values from the config right into the SQL. Please use ``?`` placeholder instead and pass it as a lists of strings. If necessary, you can use ``namedParameterJdbcTemplate`` instead of ``jdbc``, but I believe ``query()`` should be able to work with the list params.

Reply:
Oh, thank you, it should be fixed now, although I was unable to make it work without `namedParameterJdbcTemplate `.